Loop/Managers/LoopDataManager.swift

```diff
@@ -1078,6 +1096,104 @@ extension LoopDataManager {
         return prediction
     }
 
+    fileprivate func predictGlucoseFromManualGlucose(
+        _ glucose: NewGlucoseSample,
+        potentialBolus: DoseEntry?,
+        potentialCarbEntry: NewCarbEntry?,
+        replacingCarbEntry replacedCarbEntry: StoredCarbEntry?,
+        includingPendingInsulin: Bool
+    ) throws -> [PredictedGlucoseValue] {
+        let retrospectiveStart = glucose.date.addingTimeInterval(-retrospectiveCorrection.retrospectionInterval)
+        let earliestEffectDate = Date(timeIntervalSinceNow: .hours(-24))
+        let nextEffectDate = insulinCounteractionEffects.last?.endDate ?? earliestEffectDate
+
+        let updateGroup = DispatchGroup()
+        let effectCalculationError = Locked<Error?>(nil)
+
+        var insulinEffect: [GlucoseEffect]?
+        let basalDosingEnd = includingPendingInsulin ? nil : Date()
+        updateGroup.enter()
+        doseStore.getGlucoseEffects(start: nextEffectDate, basalDosingEnd: basalDosingEnd) { result in
```

Comment:
This is subtle; pending insulin can be negative. If negative pending insulin is included when computing a bolus recommendation, the recommendation may be compensating for the future withholding of insulin, resulting in a system that's competing against itself. This is a change we made when incorporating pending insulin directly into the prediction.

Reply:
Ah, I see my confusion, `getPendingInsulin` does not count negative pending temp basal.

I think my confusion stems from the fact that `getPendingInsulin` isn't really returning pending insulin. Furthermore, since this function is only *ever* compared against > 0, the better long term choice might be to rename `getPendingInsulin` to `shouldlncludePendingInsulin` and include the comparison (> 0) within the function.

Just a thought...